app/main.py

```diff
@@ -1,13 +1,36 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
+    return {
+        "line": error["line_number"] if "line_number" in error else None,
+        "column": error["column_number"] if "column_number" in error else None,
+        "message": error["text"] if "text" in error else None,
+        "name": error["code"] if "code" in error else None,
+        "source": "flake8"
+    }
     pass
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
+    return {
+        "errors":
+            [
+                format_linter_error(error)
+                for error in errors
+            ],
+        "path": file_path,
+        "status": "failed" if errors != 0 else "passed"
+    }
     pass
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
+    return [
+        {
+            "errors": [],
+            "path": file_path,
+            "status": "passed"
+        }
+        if errors == []
+        else format_single_linter_file(file_path, errors)
+        for file_path, errors in linter_report.items()
+    ]
```

Comment:
No need to check for errors, you already have to check inside format_linter_error function
There should be something like that here:
```suggestion
    return [
        format_single_linter_file(file_path, errors) for file_path,
        errors in linter_report.items()
    ]
```